### PR TITLE
Fix[RL-worker]: Updated API instance for Nvidia driver

### PR DESCRIPTION
**Pull request by Ayukha**
**Why do we need it?**
To enable the GPU machines to be used by the kubernetes pods

**Changes Proposed:**
Updated the API instance used to deploy the Nvidia-k8s-Driver

### CODE REVIEW

File: scripts/workers/rl_submission_worker.py [hunk 1]
```diff
@@ -164,9 +164,26 @@ def get_api_object(cluster_name, cluster_endpoint, challenge, evalai):
     return api_instance
 
 
-def get_core_v1_api_object(cluster_name, challenge, evalai):
+def get_instance(cluster_name, cluster_endpoint, challenge, evalai):
```
Comment (RishabhJain2018) [inline scripts/workers/rl_submission_worker.py:167]: Can we please have a meaningful name for this function?

File: scripts/workers/rl_submission_worker.py [hunk 2]
```diff
@@ -267,6 +284,7 @@ def install_gpu_drivers(api_instance):
         ext_client.create_namespaced_daemon_set(namespace, daemonset_spec)
     except ApiException as e:
         if e.status == 409:
+            print("done")
```
Comment (RishabhJain2018) [inline scripts/workers/rl_submission_worker.py:287]: Please remove the print staement.

File: scripts/workers/rl_submission_worker.py (hunk 1 above)
Comment (Ayukha) [reply scripts/workers/rl_submission_worker.py]: @RishabhJain2018  Done

File: scripts/workers/rl_submission_worker.py [hunk 3]
```diff
@@ -164,9 +164,26 @@ def get_api_object(cluster_name, cluster_endpoint, challenge, evalai):
     return api_instance
 
 
-def get_core_v1_api_object(cluster_name, challenge, evalai):
+def get_api_client(cluster_name, cluster_endpoint, challenge, evalai):
```
Comment (RishabhJain2018) [inline scripts/workers/rl_submission_worker.py:167]: @Ayukha Can we please create a single function for getting all the three API client objects.

The input to the function will be `cluster_name, cluster_endpoint, challenge, evalai, api_client_type` and you will return the api_client based upon `api_client_type`

File: scripts/workers/nvidi_device_plugin.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop: ["ALL"]
+        volumeMounts:
+          - name: device-plugin
+            mountPath: /var/lib/kubelet/device-plugins
+      volumes:
+        - name: device-plugin
+          hostPath:
+            path: /var/lib/kubelet/device-plugins
```
Comment (RishabhJain2018) [inline scripts/workers/nvidi_device_plugin.yml:58]: Please add a blank line here.

File: scripts/workers/rl_submission_worker.py (hunk 3 above)
Comment (Ayukha) [reply scripts/workers/rl_submission_worker.py:167]: Sure.

Comment (RishabhJain2018) [reply scripts/workers/rl_submission_worker.py:167]: @Ayukha Is this change done?
cc: @vkartik97 @rishabhbhardwaj

**Comment by deshraj** [comment]
@Ayukha any updates on this? Or what is the plan?

**Comment by Ayukha** [comment]
@deshraj The PR for environment logs needed few more API instances, I was thinking of updating this, once that gets merged, then I can create a common function for all API instances.

**Comment by deshraj** [comment]
@Ayukha sounds good. Can you please link the other PR here that you are talking about?

**Comment by Ayukha** [comment]
@deshraj Sure. The PR #2910

**Comment by RishabhJain2018** [comment]
@Ayukha It would be nice we can have a PR description to answer why we added this change.

**Comment by RishabhJain2018** [comment]
@Ayukha Can you please add PR description.

**Comment by Ayukha** [comment]
@RishabhJain2018  Done.

**Pull request merged**